src/Controller/TwitterController.php

```diff
@@ -50,15 +50,27 @@ public function getRequestToken(Request $request, PDO $db): array
         $res = $client->post('oauth/request_token');
 
         if ($res->getStatusCode() != Http::OK) {
-            throw new Exception("Twitter: no request token (" . $res->getStatusCode() . ": " . $res->getBody() . ")", Http::INTERNAL_SERVER_ERROR);
+            throw new Exception("Twitter: no request token (" . $res->getStatusCode() . ": " . $res->getBody() . ")", Http::SERVICE_UNAVAILABLE);
         }
 
         parse_str($res->getBody(), $data);
 
         $requestTokenMapper = new TwitterRequestTokenMapper($db, $request);
         // $tokens is instance of TwitterRequestTokenModelCollection
-        $tokens      = $requestTokenMapper->create((string) $data['oauth_token'], (string) $data['oauth_token_secret']);
-        $output_list = $tokens->getOutputView($request);
+        $oauth_token        = $data['oauth_token'];
+        $oauth_token_secret = $data['oauth_token_secret'];
+        if (!is_string($oauth_token) || !is_string($oauth_token_secret)) {
+            throw new Exception('Invalid type for oauth token or oauth secret');
+        }
+
+        $tokens = $requestTokenMapper->create($oauth_token, $oauth_token_secret);
+
+        if (!$tokens) {
```

Comment:
Can we throw in the mapper on this as well? Also, we shouldn't return a 200 with an empty list if the token wasn't created.

Reply:
should be resolved